modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CProductLocalServiceImpl.java

```diff
@@ -68,8 +71,7 @@ public CProduct deleteCProduct(CProduct cProduct) throws PortalException {
 
 		// Commerce product definitions
 
-		cpDefinitionLocalService.deleteCPDefinitions(
-			cProduct.getCProductId(), WorkflowConstants.STATUS_ANY);
+		cpDefinitionPersistence.removeByCProductId(cProduct.getCProductId());
```

Comment:
Use a new method into CPDefinitionLocalService in which you call `cpDefinitionPersistence.findByCProductId` and iterate through the list

Reply:
```suggestion
		int cpDefinitionsCount = cpDefinitionPersistence.countByCProductId(
			cpDefinition.getCProductId());

		if (cpDefinitionsCount == 1) {
			cProductLocalService.deleteCProduct(cpDefinition.getCProductId());
		}
```

Reply:
`int cpDefinitionsCount = cpDefinitionPersistence.countByCProductId(
			cpDefinition.getCProductId());

		if (cpDefinitionsCount == 1) {
			cProductLocalService.deleteCProduct(cpDefinition.getCProductId());
		}`

Reply:
```
@Override
	public void removeByCProductId(long CProductId) {
		for (CPDefinition cpDefinition :
				findByCProductId(
					CProductId, QueryUtil.ALL_POS, QueryUtil.ALL_POS, null)) {

			remove(cpDefinition);
		}
	}
```

Reply:
that's what the persistence is doing